crates/agentgateway/src/http/auth.rs

```diff
@@ -211,6 +211,29 @@ mod gcp {
 		trace!("attached GCP token");
 		Ok(hv)
 	}
+
+	// Id token for gcp audience is required. Use metadata server.
+	pub async fn get_id_token_for_audience(audience: &str) -> anyhow::Result<HeaderValue> {
+		use reqwest::Client;
+		let url = format!(
+			"http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?audience={}",
```

Comment:
I don't think `metadata` works in many environments. Maybe a CloudRun specific thing but not general, and even for cloudrun we would want calling from <notcloudrun> to cloudrun.

Would be best to utilize the existing google_cloud_auth library (even though I think its among the worst APIs I have ever used) to avoid needing to worry about details like this.